src/appleseed.studio/mainwindow/mainwindow.cpp

```diff
@@ -1254,6 +1254,9 @@ void MainWindow::start_rendering(const RenderingMode rendering_mode)
         rendering_mode == RenderingMode::InteractiveRendering ? "interactive" : "final";
     const ParamArray params = get_project_params(configuration_name);
 
+    if (!project->has_texture_store())
```

Comment:
Having to manually initialize the texture store before a render is a big problem. In fact, you haven't updated all the other places where we start a render, meaning that rendering there is now broken (appleseed.cli, appleseed.bench, Max plugin, Maya plugin, Blender plugin, Gaffer integration and possibly many others).

Reply:
One way to not manually initialize the TextureStore is to treat it analogous to TraceContext i.e. initialize the first time you need it and then use a reference everywhere else.

I'll try to implement this solution.

If you have something better in mind, please let me know.